src/ucp/api/ucp.h

```diff
@@ -732,7 +764,25 @@ typedef enum {
      * data by calling @ref ucp_am_recv_data_nbx routine. This flag is mutually
      * exclusive with @a UCP_AM_RECV_ATTR_FLAG_DATA.
      */
-    UCP_AM_RECV_ATTR_FLAG_RNDV         = UCS_BIT(17)
+    UCP_AM_RECV_ATTR_FLAG_RNDV          = UCS_BIT(17),
+
+    /**
+     * Indicates that the incoming data is the first fragment of the
+     * multi-fragment eager message. This flag can only be passed to data
+     * handlers registered without @a UCP_AM_FLAG_WHOLE_MSG flag. This flag is
+     * mutually exclusive with @a UCP_AM_RECV_ATTR_FLAG_RNDV and
+     * @a UCP_AM_RECV_ATTR_FLAG_ONLY flags.
+     */
+    UCP_AM_RECV_ATTR_FLAG_FIRST         = UCS_BIT(18),
```

Comment:
let's add `_FRAG_` to names of related flags

Reply:
maybe allow user to pass a custom AM callback to ucp_am_recv_data? so handler of nonfirst message could be potentially a different function. this could simplify the code and reduce 1 branch
also can check offset > 0?

Reply:
imo, introducing one more entity will increase overall confusion